### PR TITLE
net/netif: move gnrc_netif_addr_to/from_str() to netif

### PR DESCRIPTION
**Pull request by benpicco**
### Contribution description

These functions are independent from GNRC and can be used by stack-agnistoc code.
Avoid pulling in a GNRC dependency by moving those two helper functions to `netif`.

The old function names are kept as `static inline` wrapper functions to avoid breaking API users.

### Testing procedure

Green Murdock

### Issues/PRs references

avoids GNRC dependency in #14758

### CODE REVIEW

**Comment by maribu** [comment]
Murdock failed:

```
[INFO] patch stm32cmsis
fatal: previous rebase directory .git/rebase-apply still exists but mbox given.
/tmp/dwq.0.7514119904954071/3db93550341a72ee6a1fea3d05e019f6/pkg/pkg.mk:107: recipe for target '/tmp/dwq.0.7514119904954071/3db93550341a72ee6a1fea3d05e019f6/cpu/stm32/include/vendor/cmsis/l4/.pkg-state.git-patched' failed
make[1]: *** [/tmp/dwq.0.7514119904954071/3db93550341a72ee6a1fea3d05e019f6/cpu/stm32/include/vendor/cmsis/l4/.pkg-state.git-patched] Error 128
/tmp/dwq.0.7514119904954071/3db93550341a72ee6a1fea3d05e019f6/tests/cond_order/../../Makefile.include:686: recipe for target 'pkg-prepare' failed
```

Could it be that patches are applied in parallel when `make -j` is used? Or maybe `build/pkg` is stored on a shared network file system and when two STM32 boards get being build in parallel on two workers, those two interfere with preparing the package?

@kaspar030 might know this.

**Comment by miri64** [comment]
> @kaspar030 might know this.

Or @aabadie?

**Comment by benpicco** [comment]
This happens rather quite a bit, running Murdock again will usually fix it.
There is either a race condition in the build system or GitHub is rate-limiting is it seems.

**Comment by maribu** [comment]
The issue above comes when you try to apply a patch with `git apply` while git is already applying a patch. There is no network operation involved.

So this is indeed a race condition. But the question is whether we can locally reproduce it (using `make -j`), or if it is indeed only affecting the CI.

**Comment by cgundogan** [comment]
>There is either a race condition in the build system or GitHub is rate-limiting is it seems.

@leandrolanzieri might have some insights on this?

**Comment by benpicco** [comment]
IIRC CI was only using one thread / worker.

File: tests/netdev_test/main.c [hunk 1]
```diff
@@ -198,11 +198,11 @@ static int test_receive(void)
         puts("=================");
         puts("expected");
         puts("=================");
-        puts(gnrc_netif_addr_to_str(_test_src, ETHERNET_ADDR_LEN, addr_str));
+        puts(netif_addr_to_str(_test_src, ETHERNET_ADDR_LEN, addr_str));
         puts("=================");
         puts("received source");
         puts("=================");
-        puts(gnrc_netif_addr_to_str(gnrc_netif_hdr_get_src_addr(hdr->data),
+        puts(netif_addr_to_str(gnrc_netif_hdr_get_src_addr(hdr->data),
```
Comment (miri64) [inline tests/netdev_test/main.c:205]: I said keep the old tests... I would prefer a dedicated test application using `embUnit` for this.

**Comment by fjmolinas** [comment]
All green @miri64!

**Comment by miri64** [comment]
There still is no separate test application for the new functionality as far as I can see.

**Comment by benpicco** [comment]
`tests/gnrc_netif` also had tests for common `netif_` functions, so I thought that would be the most logical place to put it.

**Comment by miri64** [comment]
> `tests/gnrc_netif` also had tests for common `netif_` functions, so I thought that would be the most logical place to put it.

You argued for less dependencies for the conversion functions, so I think the most logical place to put the tests is in a separate test application that _just_ uses the `netif` dependency ;-).

**Comment by benpicco** [comment]
I'm arguing for less dependencies for users of the API, not tests :wink: 

But if you want to give me (and Murdock) more work for the sake of orderliness so be it.

Should the other `test_netif_` functions be moved too?

**Comment by miri64** [comment]
> I'm arguing for less dependencies for users of the API, not tests :wink:

But that should be tested as well IMHO

> But if you want to give me (and Murdock) more work for the sake of orderliness so be it.

If we start argue like that then we should not allow for any compile time optimization or alternative implementations, as those would be untestable as well under these restrictions ;-).

> Should the other `test_netif_` functions be moved too?

If they are testing function implemented in `netif`, you may (but can as well be done in a follow-up). If they are only abstract functions that are required to be implemented in the stack, they should remain with the network stack's tests.

**Comment by benpicco** [comment]
I moved the testcase to `tests/netif`.

**Review by miri64 — APPROVED** [review]
ACK. I confirmed, that this mainly just moves code around. To test I ran both `tests/netif` and `tests/gnrc_netif` on `native` and `samr21-xpro`. Both still pass.

**Comment by miri64** [comment]
Please squash!

**Pull request merged**